### ISSUE TITLE
DiskLruCache is not main-safe - performs file i/o on the current thread.

### ISSUE THREAD

**Issue by prashanOS** [apollographql/apollo-kotlin#4602]
### Version

3.7.3

### Summary

Currently, DiskLruCache + HttpCacheExtensions leads to an Android StrictMode DiskReadViolation (which occurs when file I/O is performed on the main thread) when entries are removed from the cache:

Example from HttpCacheExtensions:
`cacheKey?.let { cachingHttpInterceptor.cache.remove(it) }`

DiskLruCache performs file I/O and therefore either needs to internally make sure this work is happening on a background thread or consumers (in this case HttpCacheExtensions) must enforce this. Neither is being done right now.

If you don't have it set up already, would highly recommend turning on StrictMode in CI for your Android project so that you can catch these issues in testing!

### Steps to reproduce the behavior

1. Turn on DiskLruCache.
2. Enable Android StrictMode with disk read violations set to fatal.
3. Make a query that will return a response with errors to trigger cache removal.
4. See crash.

### Logs

```
Process: <omitted>, PID: 19851
java.lang.RuntimeException: StrictMode ThreadPolicy violation
        at android.os.StrictMode$AndroidBlockGuardPolicy.onThreadPolicyViolation(StrictMode.java:1875)
	at android.os.StrictMode$AndroidBlockGuardPolicy.lambda$handleViolationWithTimingAttempt$0$android-os-StrictMode$AndroidBlockGuardPolicy(StrictMode.java:1789)
	at android.os.StrictMode$AndroidBlockGuardPolicy$$ExternalSyntheticLambda1.run(Unknown Source:6)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7872)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
Caused by: android.os.strictmode.DiskReadViolation
	at android.os.StrictMode$AndroidBlockGuardPolicy.onReadFromDisk(StrictMode.java:1658)
	at libcore.io.BlockGuardOs.open(BlockGuardOs.java:269)
	at libcore.io.ForwardingOs.open(ForwardingOs.java:563)
	at android.app.ActivityThread$AndroidOs.open(ActivityThread.java:7758)
	at libcore.io.IoBridge.open(IoBridge.java:560)
	at java.io.FileInputStream.<init>(FileInputStream.java:160)
	at okio.Okio__JvmOkioKt.source(JvmOkio.kt:182)
	at okio.Okio.source(Unknown Source:1)
	at okio.JvmSystemFileSystem.source(JvmSystemFileSystem.kt:95)
	at com.apollographql.apollo3.cache.http.internal.DiskLruCacheKt.source(DiskLruCache.kt:1020)
	at com.apollographql.apollo3.cache.http.internal.DiskLruCacheKt.access$source(DiskLruCache.kt:1)
	at com.apollographql.apollo3.cache.http.internal.DiskLruCache.readJournal(DiskLruCache.kt:231)
	at com.apollographql.apollo3.cache.http.internal.DiskLruCache.initialize(DiskLruCache.kt:208)
	at com.apollographql.apollo3.cache.http.internal.DiskLruCache.remove(DiskLruCache.kt:545)
	at com.apollographql.apollo3.cache.http.DiskLruHttpCache.remove(DiskLruHttpCache.kt:103)
	at com.apollographql.apollo3.cache.http.HttpCache$httpCache$2$intercept$1$2.invokeSuspend(HttpCacheExtensions.kt:130)
	at com.apollographql.apollo3.cache.http.HttpCache$httpCache$2$intercept$1$2.invoke(Unknown Source:8)
	at com.apollographql.apollo3.cache.http.HttpCache$httpCache$2$intercept$1$2.invoke(Unknown Source:4)
        <rest of stack trace omitted for brevity>
```

**Comment by BoD** [comment]
Thanks for reporting this! A fix is in #4606.

**Comment by martinbonnin** [comment]
This is now available in [version 3.7.4](https://github.com/apollographql/apollo-kotlin/releases/tag/v3.7.4)
